### PR TITLE
Update to React 0.14

### PR DESCRIPTION
**Pull request by danvk**
Highlights:
- `React` → `ReactDOM`
- Drop many `.getDOMNode()` calls
- `React.addons.classSet` → `require('classnames')`
- `React.addons.TestUtils` → `require('react-addons-test-utils')`
- `ReactClass.type` → `ReactClass`
- Updated to `pileup@0.5.1`

[Review on Reviewable](https://reviewable.io/reviews/hammerlab/cycledash/858)

### CODE REVIEW

File: cycledash/static/js/runs/components/RunsPage.js [hunk 1]
```diff
@@ -125,13 +126,11 @@ var RunsPage = React.createClass({
...
           </div>
           <div className="projects-table">
-          <div className='filter-runs'>
-            <label>Now Viewing</label>
-            <select className='select-project-filter' value={this.state.projectFilter}
-                    onChange={this.handleProjectFilter}>
-              {projectOptions}
-            </select>
-          </div>
+            <div className='filter-runs'>
```
Comment (ihodes) [inline cycledash/static/js/runs/components/RunsPage.js]: Doesn't seem like this file should be changed?

**Comment by ihodes** [comment]
Thanks for debugging this. Is it worth doing this in two steps? Seems like the 0.13 change is just a massive regression in terms of functionality, would be better to do 0.12→0.14. Are you planning on upgrading this to 0.14 right away?

File: cycledash/static/js/runs/components/RunsPage.js (hunk 1 above)
Comment (danvk) [inline cycledash/static/js/runs/components/RunsPage.js]: Yeah, just an indentation fix. Reverted.

**Comment by danvk** [comment]
Yeah, good point. I'll update to 0.14 and push another commit to this branch.

**Comment by ihodes** [comment]
Cool, sounds good, thanks!

**Comment by danvk** [comment]
This depends on the next version of pileup (see https://github.com/hammerlab/pileup.js/issues/246), which I hope to release tomorrow.

**Comment by ihodes** [comment]
LGTM, thanks for doing this!

**Pull request merged**